### PR TITLE
Allow testing p3d versions

### PR DESCRIPTION
**Pull request by tannermpeterson**

### CODE REVIEW

File: electron/main/index.js [hunk 1]
```diff
@@ -153,11 +153,9 @@ ipcMain.once("pulse3d_versions_request", (event) => {
   axios
     .get(`https://${CLOUD_PULSE3D_ENDPOINT}/versions`)
     .then((response) => {
-      const versions = response.data
-        .filter(({ state }) => state === "external")
-        .map(({ version }) => version);
```
Comment (roberto-araiza-curibio) [inline electron/main/index.js:158]: qq: we don't do this `.map` here (1/2)

File: ui/components/settings/SettingsForm.vue [hunk 2]
```diff
@@ -242,16 +243,37 @@ export default {
...
+        mapped_versions[info.version] = info;
+      });
+      const external_versions = semver_sort.desc(
+        this.pulse3d_version_info.filter(({ state }) => state === "external").map(({ version }) => version)
+      );
+      const testing_versions = semver_sort.desc(
+        this.pulse3d_version_info.filter(({ state }) => state === "testing").map(({ version }) => version)
+      );
+      const deprecated_versions = semver_sort.desc(
+        this.pulse3d_version_info.filter(({ state }) => state === "deprecated").map(({ version }) => version)
```
Comment (roberto-araiza-curibio) [inline ui/components/settings/SettingsForm.vue:266]: because we're doing these `.map`s here? (2/2)

**Review by roberto-araiza-curibio — APPROVED** [review]
LGTM.  One minor question (in two parts) below.

File: ui/components/settings/SettingsForm.vue (hunk 2 above)
Comment (tannermpeterson) [reply ui/components/settings/SettingsForm.vue:266]: Yes, we need to store all the information that comes back from that response and filter later

**Pull request merged**